server/api/crud/model_monitoring/deployment.py

```diff
@@ -980,11 +980,10 @@ def _delete_model_monitoring_stream_resources(
                     logger.debug(
                         "Deleted v3io stream", project=project, stream_path=stream_path
                     )
-                except v3io.dataplane.response.HttpResponseError as e:
-                    logger.warning(
-                        "Failed to delete v3io stream",
-                        stream_path=stream_path,
-                        error=mlrun.errors.err_to_str(e),
+                except Exception as exc:
+                    # Raise an error that will be caught by the caller and skip the deletion of the stream
+                    raise mlrun.errors.MLRunStreamConnectionFailure(
+                        f"Failed to delete v3io stream {stream_path}, {mlrun.errors.err_to_str(exc)}"
```

Comment:
Should we use from exc here, to include the original stack trace?

Reply:
Why do we have to repackage the exception anyway? Why not just let it propagate up the stack?

Reply:
> Why do we have to repackage the exception anyway? Why not just let it propagate up the stack?

We want to wrap this part in the **`MLRunStreamConnectionFailure`** error, which we know the caller can handle. If a different error occurs before it, we may want the process to fail.